types/react-redux/react-redux-tests.tsx

```diff
@@ -357,22 +357,80 @@ namespace TestTOwnPropsInference {
         return { state: 'string' };
     }
 
-    const ConnectedWithoutOwnProps = connect(mapStateToPropsWithoutOwnProps)(OwnPropsComponent);
-    const ConnectedWithOwnProps = connect(mapStateToPropsWithOwnProps)(OwnPropsComponent);
-    const ConnectedWithTypeHint = connect<StateProps, void, OwnProps>(mapStateToPropsWithoutOwnProps)(OwnPropsComponent);
+    // `OwnProps` can not be inferred, which is good
+    // const ConnectedWithoutOwnProps = connect(mapStateToPropsWithoutOwnProps)(OwnPropsComponent);
+    // Otherwise, this compiles, which is bad.
+    // React.createElement(ConnectedWithoutOwnProps, { anything: 'goes!' });
```

Comment:
this test was exposing a bug where incorrect parameters would compile.
`ConnectedWithoutOwnProps` can no longer infer typing, invalidating this test case